### PR TITLE
File boards files not being deleted properly

### PR DESCRIPTION
**Pull request by crspeller**
#### Summary

Fixes deletion of files attachments when cards are deleted.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-51879

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: server/boards/app/blocks.go [hunk 1]
```diff
@@ -408,25 +408,39 @@ func (a *App) DeleteBlockAndNotify(blockID string, modifiedBy string, disableNot
...
 
-	err = a.store.DeleteBlock(blockID, modifiedBy)
-	if err != nil {
-		return err
-	}
-
 	if block.Type == model.TypeImage {
 		fileName, fileIDExists := block.Fields["fileId"]
 		if fileName, fileIDIsString := fileName.(string); fileIDExists && fileIDIsString {
 			filePath := filepath.Join(block.BoardID, fileName)
```
Comment (sbishel) [inline server/boards/app/blocks.go:414]: I realize this line existed previously, but I'm not sure this works. I believe "teamID" needs to be included as well. 
https://github.com/mattermost/mattermost-server/blob/master/server/boards/app/files.go#L104

File: server/boards/app/blocks.go [hunk 2]
```diff
@@ -408,25 +408,39 @@ func (a *App) DeleteBlockAndNotify(blockID string, modifiedBy string, disableNot
 		return nil
 	}
 
-	err = a.store.DeleteBlock(blockID, modifiedBy)
-	if err != nil {
-		return err
-	}
-
 	if block.Type == model.TypeImage {
```
Comment (sbishel) [inline server/boards/app/blocks.go:411]: This section will also need to check for the FileInfo row, similar to Attachments. If no fileInfo row, then create path and delete.

**Review by sbishel — COMMENTED** [review]
We may want to merge to this PR. 

https://github.com/mattermost/mattermost-server/pull/22915

**Comment by crspeller** [comment]
Will re-open with new solution.

**Pull request closed without merging**